MainWindow.cpp

```diff
@@ -125,13 +125,15 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), _ui(new Ui::MainW
   this->_recentFiles = new RecentFiles(this, this->_ui->menuRecent, this->_ui->actionClearRecentMenu);
 
   _ui->mdiArea->setBackground(QImage(":/banner.png"));
+  connect(_ui->mdiArea, &QMdiArea::subWindowActivated, this, &MainWindow::MDIWindowChanged);
   connect(_ui->menuWindow, &QMenu::aboutToShow, this, &MainWindow::updateWindowMenu);
 
   auto settingsButton = static_cast<QToolButton *>(_ui->mainToolBar->widgetForAction(_ui->actionSettings));
   settingsButton->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
   settingsButton->setToolButtonStyle(Qt::ToolButtonStyle::ToolButtonTextBesideIcon);
   _ui->actionSettings->setMenu(_ui->menuChangeGameSettings);
 
+#ifdef RGM_SERVER_ENABLED

```

Comment:
Since you're linking in all the code regardless of whether you intend to use it, this could as well be a runtime check.

Also, we're leaking that pluginServer ref. It should probably be a `unique_ptr` owned by this class (can be left null when disabled).

Reply:
Are you suggesting making it a dll? Yeah that's how Qt Creator has its own plugin interface, I just wasn't ready to set it up yet.

And no, we're not, Qt has its own ownership policy. The server plugin reference is owned by the main window. This is because the server plugin inherits from my generic RGM plugin abstraction which itself is based on QObject. All RGM plugins are informed of the main window at the present time.
https://github.com/enigma-dev/RadialGM/blob/6d626d725f3132f829d54028a5556abb38d89a09/Plugins/RGMPlugin.h#L31

Reply:
No, I'm just suggesting being consistent about whether we're eliminating code at compile time or branching over it at runtime. My complaint above is related.

> I am checking the define `RGM_SERVER_ENABLED` in `MainWindow.cpp` which calls this constructor.

You are building an entire unused class into RGM in this case. I suggest either making it a runtime check (e.g, a flag), or removing the class from the codegen.

Reply:
Which class is the one you are saying is unused? Are you saying RGMPlugin is unused or ServerPlugin is unused? The former is clearly true, but the latter is not always true. I am in fact filtering out the ServerPlugin class entirely in both the pro qmake and the CMakeLists.

Reply:
So if you remove the `#if` from just MainWindow, it will generate link errors because the Qt project file excludes the class?

Reply:
Yes, assuming you haven't uncommented `rgm_enable_grpc_server` in the pro or CMakeLists.

Reply:
A bit awkward, but okay. Proceed.

Reply:
Silly me though I forgot to wrap the include in the main window.

Reply:
It's ok though, as I said later we'll move plugins to dlls.